docs/grpc_disruption/interceptor_algorithm.md

```diff
@@ -0,0 +1,180 @@
+# gRPC Disruption Inceptor
+
+When the interceptor recognizes a query's endpoint as one which is actively getting disrupted, the interceptor generates a random integer from `0` to `100`, and consults a `PercentToAlteration` mapping to figure out what `alteration` to apply to a query response. This mapping is populated by the percentage odds a user configured for each alteration. Currently, we support two `alteration`s:
+
+1. return a gRPC error code (such as `NotFound` or `PermissionDenied`)
+2. return an empty response (`emptypb.Empty`)
+
+You can see an example below of a mapping that does not define all 100% of possible requests below.
+
+## gRPC Disruption - Algorithm Examples
+
+### Multiple alterations with defined query percentages
+
+The following is a complex example to illustrate the algorithm's behavior.
+
+```
+spec:
+  grpc:
+    endpoints:
+      - endpoint: /chaos_dogfood.ChaosDogfood/order
+        override: "{}"
+        query_pct: 5
+      - endpoint: /chaos_dogfood.ChaosDogfood/order
+        error: NOT_FOUND
+        query_pct: 5
+      - endpoint: /chaos_dogfood.ChaosDogfood/order
+        error: PERMISSION_DENIED
+        query_pct: 15
+```
+
+For the above specs, the calculated `PercentToAlteration` would look something like:
+
+```
+PercentToAlteration {
+    0  -> Override: {}
+    1  -> Override: {}
+    2  -> Override: {}
+    3  -> Override: {}
+    4  -> Override: {}
+    5  -> Error: NOT_FOUND
+    6  -> Error: NOT_FOUND
+    7  -> Error: NOT_FOUND
+    8  -> Error: NOT_FOUND
+    9  -> Error: NOT_FOUND
+    10 -> Error: PERMISSION_DENIED
+    11 -> Error: PERMISSION_DENIED
+    12 -> Error: PERMISSION_DENIED
+    13 -> Error: PERMISSION_DENIED
+    .. -> Error: ...
+    22 -> Error: PERMISSION_DENIED
+    23 -> Error: PERMISSION_DENIED
+    24 -> Error: PERMISSION_DENIED
+}
+```
+
+In this case, we would return an Override with empty results for 5% of queries, a `NOT_FOUND` error for 5% of queries, and return `PERMISSION_DENIED` for 15% of queries.
+
+:warning: If you are a maintainer looking at the source code, the "mapping" is represented as a golang slice with length <= 100.
```

Comment:
It seems strange to see documentation here aimed at the maintainer. Maybe you can move this explanation somewhere like a comment in your code (which I assume you already have) or maybe a clever way of doing it is referencing the code somewhere here where a comment saying this is located.